[java] Fury not support serialization of private interface lambda ?

fury version: 0.1.1
jdk version: 11

When I am serializing the code below
```java
public class TestClass implements Serializable {
    private interface A extends Serializable {
        void t();
    }
    private A a = () -> {
    };
}
```

> java.lang.RuntimeException: java.lang.RuntimeException: Create compatible serializer failed, 
> class: class link.yxw.entity.custom.TestClass
>
> 	at link.yxw.marshaller.fury.FuryMarshaller.serialize(FuryMarshaller.java:36)
> 	at link.yxw.DataTest.testDataMarshaller(DataTest.java:49)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
> 	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
> 	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
> 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
> 	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
> 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
> 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestTemplateMethod(TimeoutExtension.java:94)
> 	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
> 	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
> 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
> 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
> 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
> 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
> 	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
> 	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
> 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
> 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
> 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
> 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
> 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
> 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
> 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
> 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
> 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask$DefaultDynamicTestExecutor.execute(NodeTestTask.java:226)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask$DefaultDynamicTestExecutor.execute(NodeTestTask.java:204)
> 	at org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor.execute(TestTemplateTestDescriptor.java:142)
> 	at org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor.lambda$execute$2(TestTemplateTestDescriptor.java:110)
> 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
> 	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
> 	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:274)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
> 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
> 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
> 	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
> 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
> 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
> 	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
> 	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:274)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
> 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
> 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
> 	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
> 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
> 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
> 	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
> 	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:274)
> 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
> 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
> 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
> 	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
> 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
> 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
> 	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
> 	at org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor.execute(TestTemplateTestDescriptor.java:110)
> 	at org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor.execute(TestTemplateTestDescriptor.java:44)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
> 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
> 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
> 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
> 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
> 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
> 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
> 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
> 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
> 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
> 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
> 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
> 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
> 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
> 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
> 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
> 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
> 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
> 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
> 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
> 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
> 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
> 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
> 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
> 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
> 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
> 	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
> 	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
> 	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:57)
> 	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
> 	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
> 	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
> 	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:232)
> 	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:55)
> Caused by: java.lang.RuntimeException: Create compatible serializer failed, 
> class: class link.yxw.entity.custom.TestClass
> 	at io.fury.serializer.CodegenSerializer.loadCompatibleCodegenSerializer(CodegenSerializer.java:56)
> 	at io.fury.resolver.ClassResolver.lambda$getObjectSerializerClass$4(ClassResolver.java:872)
> 	at io.fury.builder.JITContext.registerSerializerJITCallback(JITContext.java:132)
> 	at io.fury.resolver.ClassResolver.getObjectSerializerClass(ClassResolver.java:867)
> 	at io.fury.resolver.ClassResolver.getSerializerClass(ClassResolver.java:811)
> 	at io.fury.resolver.ClassResolver.getSerializerClass(ClassResolver.java:720)
> 	at io.fury.resolver.ClassResolver.createSerializer(ClassResolver.java:1067)
> 	at io.fury.resolver.ClassResolver.getOrUpdateClassInfo(ClassResolver.java:1009)
> 	at io.fury.Fury.writeRef(Fury.java:282)
> 	at io.fury.Fury.serialize(Fury.java:234)
> 	at io.fury.Fury.serialize(Fury.java:253)
> 	at io.fury.Fury.serialize(Fury.java:247)
> 	at link.yxw.marshaller.fury.FuryMarshaller.serialize(FuryMarshaller.java:32)
> 	... 127 more
> Caused by: java.lang.IllegalArgumentException: Field type should be public for codegen-based access
> 	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:143)
> 	at io.fury.builder.CodecBuilder.getFieldValue(CodecBuilder.java:110)
> 	at io.fury.builder.CompatibleCodecBuilder.lambda$null$8f405a5f$1(CompatibleCodecBuilder.java:246)
> 	at io.fury.codegen.ExpressionOptimizer.invokeGenerated(ExpressionOptimizer.java:48)
> 	at io.fury.builder.CompatibleCodecBuilder.lambda$invokeGenerated$0(CompatibleCodecBuilder.java:148)
> 	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1134)
> 	at io.fury.builder.CompatibleCodecBuilder.invokeGenerated(CompatibleCodecBuilder.java:146)
> 	at io.fury.builder.CompatibleCodecBuilder.lambda$null$a043d8ba$4(CompatibleCodecBuilder.java:243)
> 	at io.fury.codegen.ExpressionOptimizer.invokeGenerated(ExpressionOptimizer.java:48)
> 	at io.fury.builder.CompatibleCodecBuilder.lambda$buildEncodeExpression$4(CompatibleCodecBuilder.java:229)
> 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
> 	at io.fury.builder.CompatibleCodecBuilder.buildEncodeExpression(CompatibleCodecBuilder.java:226)
> 	at io.fury.builder.BaseObjectCodecBuilder.genCode(BaseObjectCodecBuilder.java:182)
> 	at io.fury.codegen.CompileUnit.getCode(CompileUnit.java:54)
> 	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:73)
> 	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:65)
> 	at io.fury.codegen.CodeGenerator.compile(CodeGenerator.java:141)
> 	at io.fury.builder.CodecUtils.loadOrGenCodecClass(CodecUtils.java:89)
> 	at io.fury.builder.CodecUtils.loadOrGenCompatibleCodecClass(CodecUtils.java:62)
> 	at io.fury.builder.CodecUtils.loadOrGenCompatibleCodecClass(CodecUtils.java:54)
> 	at io.fury.serializer.CodegenSerializer.loadCompatibleCodegenSerializer(CodegenSerializer.java:53)
> 	... 139 more

--- a/java/fury-core/src/main/java/io/fury/builder/AccessorHelper.java
+++ b/java/fury-core/src/main/java/io/fury/builder/AccessorHelper.java
@@ -205,11 +205,17 @@ public class AccessorHelper {
   }
 
   public static boolean defineAccessor(Field field) {
+    if (ReflectionUtils.isPrivate(field.getType())) {
+      return false;
+    }
     Class<?> beanClass = field.getDeclaringClass();
     return defineAccessorClass(beanClass);
   }
 
   public static boolean defineAccessor(Method method) {
+    if (ReflectionUtils.isPrivate(method.getReturnType())) {
+      return false;
+    }
     Class<?> beanClass = method.getDeclaringClass();
     return defineAccessorClass(beanClass);
   }

--- a/java/fury-core/src/main/java/io/fury/builder/CodecBuilder.java
+++ b/java/fury-core/src/main/java/io/fury/builder/CodecBuilder.java
@@ -34,6 +34,7 @@ import com.google.common.reflect.TypeToken;
 import io.fury.Fury;
 import io.fury.codegen.CodegenContext;
 import io.fury.codegen.Expression;
+import io.fury.codegen.Expression.Cast;
 import io.fury.codegen.Expression.Inlineable;
 import io.fury.codegen.Expression.Invoke;
 import io.fury.codegen.Expression.Literal;
@@ -206,7 +207,7 @@ public abstract class CodecBuilder {
                 OBJECT_TYPE,
                 beanClassExpr(),
                 Literal.ofString(fieldName));
-        getter = new Expression.Cast(getter, getterType);
+        getter = new Cast(getter, getterType);
         ctx.addField(funcInterface, key, getter);
         ref = new Reference(key, getterType);
         fieldMap.put(key, ref);
@@ -221,7 +222,7 @@ public abstract class CodecBuilder {
     Reference fieldRef = getOrCreateField(cls, descriptor.getName());
     // boolean fieldNullable = !descriptor.getTypeToken().isPrimitive();
     Invoke getObj = new Invoke(fieldRef, "get", OBJECT_TYPE, fieldNullable, inputObject);
-    return new Expression.Cast(getObj, descriptor.getTypeToken(), descriptor.getName());
+    return new Cast(getObj, descriptor.getTypeToken(), descriptor.getName());
   }
 
   /** Returns an expression that get field value> from <code>bean</code> using {@link Unsafe}. */
@@ -250,7 +251,7 @@ public abstract class CodecBuilder {
               inputObject,
               fieldOffsetExpr);
       TypeToken<?> publicSuperType = ReflectionUtils.getPublicSuperType(descriptor.getTypeToken());
-      return new Expression.Cast(getObj, publicSuperType, fieldName);
+      return new Cast(getObj, publicSuperType, fieldName);
     }
   }
 
@@ -277,6 +278,9 @@ public abstract class CodecBuilder {
       if (!Modifier.isFinal(d.getModifiers()) && !Modifier.isPrivate(d.getModifiers())) {
         if (AccessorHelper.defineAccessor(d.getField())) {
           Class<?> accessorClass = AccessorHelper.getAccessorClass(d.getField());
+          if (!value.type().equals(d.getTypeToken())) {
+            value = new Cast(value, d.getTypeToken());
+          }
           return new StaticInvoke(
               accessorClass, d.getName(), PRIMITIVE_VOID_TYPE, false, bean, value);
         }
@@ -284,6 +288,9 @@ public abstract class CodecBuilder {
       if (d.getWriteMethod() != null && !Modifier.isPrivate(d.getWriteMethod().getModifiers())) {
         if (AccessorHelper.defineAccessor(d.getWriteMethod())) {
           Class<?> accessorClass = AccessorHelper.getAccessorClass(d.getWriteMethod());
+          if (!value.type().equals(d.getTypeToken())) {
+            value = new Cast(value, d.getTypeToken());
+          }
           return new StaticInvoke(
               accessorClass, d.getWriteMethod().getName(), PRIMITIVE_VOID_TYPE, false, bean, value);
         }

--- a/java/fury-core/src/main/java/io/fury/codegen/Expression.java
+++ b/java/fury-core/src/main/java/io/fury/codegen/Expression.java
@@ -722,7 +722,7 @@ public interface Expression {
       this.castedValueNamePrefix = castedValueNamePrefix;
       inlineCall = inline;
       this.ignoreUpcast = ignoreUpcast;
-      checkArgument(ReflectionUtils.isPublic(type), "Type %s is not public", type);
+      checkArgument(!ReflectionUtils.isPrivate(type), "Type %s is private", type);
     }
 
     @Override

--- a/java/fury-core/src/main/java/io/fury/util/ReflectionUtils.java
+++ b/java/fury-core/src/main/java/io/fury/util/ReflectionUtils.java
@@ -16,6 +16,7 @@
 
 package io.fury.util;
 
+import static io.fury.type.TypeUtils.OBJECT_TYPE;
 import static io.fury.type.TypeUtils.getRawType;
 
 import com.google.common.base.Preconditions;
@@ -40,6 +41,7 @@ import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
@@ -364,17 +366,35 @@ public class ReflectionUtils {
     return Modifier.isPublic(type.getModifiers());
   }
 
+  public static boolean isPrivate(TypeToken<?> targetType) {
+    return Modifier.isPrivate(getRawType(targetType).getModifiers());
+  }
+
+  public static boolean isPrivate(Class<?> cls) {
+    return Modifier.isPrivate(cls.getModifiers());
+  }
+
   public static boolean isFinal(Class<?> targetType) {
     return Modifier.isFinal(targetType.getModifiers());
   }
 
   public static TypeToken getPublicSuperType(TypeToken typeToken) {
     if (!isPublic(typeToken)) {
-      Class<?> cls = getRawType(typeToken);
-      while (!isPublic(cls)) {
+      Class<?> rawType = Objects.requireNonNull(getRawType(typeToken));
+      Class<?> cls = rawType;
+      while (cls != null && !isPublic(cls)) {
         cls = cls.getSuperclass();
       }
-      return TypeToken.of(cls);
+      if (cls == null) {
+        for (Class<?> typeInterface : rawType.getInterfaces()) {
+          if (isPublic(typeInterface)) {
+            return TypeToken.of(typeInterface);
+          }
+        }
+        return OBJECT_TYPE;
+      } else {
+        return TypeToken.of(cls);
+      }
     } else {
       return typeToken;
     }

--- a/java/fury-core/src/test/java/io/fury/serializer/CodegenSerializerTest.java
+++ b/java/fury-core/src/test/java/io/fury/serializer/CodegenSerializerTest.java
@@ -261,16 +261,19 @@ public class CodegenSerializerTest extends FuryTestBase {
   }
 
   private interface PrivateInterface extends Serializable {
-    void t();
+    Object t();
   }
 
   public static class TestClass1 {
-    PrivateInterface a = () -> {};
+    PrivateInterface a = () -> 1;
+    private PrivateInterface b = () -> 2;
   }
 
   @Test
   public void testPrivateInterfaceField() {
     Fury fury = Fury.builder().requireClassRegistration(false).build();
-    serDe(fury, new TestClass1());
+    TestClass1 o = serDe(fury, new TestClass1());
+    Assert.assertEquals(o.a.t(), 1);
+    Assert.assertEquals(o.b.t(), 2);
   }
 }

--- a/java/fury-core/src/test/java/io/fury/serializer/CodegenSerializerTest.java
+++ b/java/fury-core/src/test/java/io/fury/serializer/CodegenSerializerTest.java
@@ -27,6 +27,7 @@ import io.fury.Language;
 import io.fury.memory.MemoryBuffer;
 import io.fury.memory.MemoryUtils;
 import io.fury.test.bean.Cyclic;
+import java.io.Serializable;
 import java.util.ArrayList;
 import java.util.List;
 import lombok.Data;
@@ -257,5 +258,19 @@ public class CodegenSerializerTest extends FuryTestBase {
     fury.register(Get.class);
     Get get = new Get(Lists.newArrayList(new Column(new byte[] {1}, new byte[] {2}, "abc")));
     serDeCheck(fury, get);
+  }
+
+  private interface PrivateInterface extends Serializable {
+    void t();
+  }
+
+  public static class TestClass1 {
+    PrivateInterface a = () -> {};
+  }
+
+  @Test
+  public void testPrivateInterfaceField() {
+    Fury fury = Fury.builder().requireClassRegistration(false).build();
+    serDe(fury, new TestClass1());
   }
 }

